src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -58,9 +58,9 @@ internal unsafe static string NormalizeDirectorySeparatorsIfNecessary(ReadOnlySp
             }
 
             if (normalized)
-                return null;
+                return path;
 
-            fixed (char* f = &MemoryMarshal.GetReference(path))
+            fixed (char* f = path)
```

Comment:
You do not need to use unsafe code and pin it anymore now. You can just pass the string through as string.